Update naas docs to call out expectations on network attachments (#1429)
nginx/documentation: content/nginxaas-google/getting-started/create-deployment/deploy-console.md

### Create a network attachment

NGINXaaS requires a [network attachment](https://cloud.google.com/vpc/docs/about-network-attachments) to connect your NGINXaaS deployment to your VPC network.

1. Access the [Google Cloud Console](https://console.cloud.google.com/).
1. Create a consumer VPC network and subnetwork. See [Google's documentation on creating a VPC and subnet](https://cloud.google.com/vpc/docs/create-modify-vpc-networks#console_1) for a step-by-step guide.

>    - The region you choose in this step **must match** the region where your NGINXaaS deployment will be created.

Reviewer: nit: do we actually want to soften this a bit here since this is docs for the UI and the UI takes care of setting the deployment region to this network attachment region? (still useful to document, but there's nothing really for the user to worry about here)

```suggestion
   - The region you choose in this step is the region where your NGINXaaS deployment will be created.
```

Author: rephrased to highlight this a bit more.

Reviewer: i was asking about highlighting it _less_

Author: I think the highlight is important as a lot of times a user will create an attachment wherever but then go to the console and find out that it may be in a region that we do not support.

Going to resolve this like we talked offline. We can revisit it if needed 👍

> 1. Create a network attachment in your new subnet. See [Google's documentation on creating a network attachment](https://cloud.google.com/vpc/docs/create-manage-network-attachments#console_1) for a step-by-step guide.

Author: I am open to adding this if we have a strong preference 👍 

P.S I also think that securing NGINXaaS could be a thing of its own where talk about things such as guarding cloud infrastructure that interact with NGINXaaS or following certain NGINX security best practices that apply to our service.

and the creation flow stays clean and minimal avoid multiple forks that a user has to pay attention to.

> 1. Create a network attachment in your new subnet. See [Google's documentation on creating a network attachment](https://cloud.google.com/vpc/docs/create-manage-network-attachments#console_1) for a step-by-step guide.

Author: Good point. I'll add a blurb around it here.

### Create a network attachment

NGINXaaS requires a [network attachment](https://cloud.google.com/vpc/docs/about-network-attachments) to connect your NGINXaaS deployment to your VPC network.

1. Access the [Google Cloud Console](https://console.cloud.google.com/).
1. Create a consumer VPC network and subnetwork. See [Google's documentation on creating a VPC and subnet](https://cloud.google.com/vpc/docs/create-modify-vpc-networks#console_1) for a step-by-step guide.
   - The region you choose in this step **must be the same** as the region where you plan to create your NGINXaaS deployment.
1. Create a network attachment in your new subnet. See [Google's documentation on creating a network attachment](https://cloud.google.com/vpc/docs/create-manage-network-attachments#create-network-attachments) for a step-by-step guide.

>    - It is recommended to configure your network attachment to manually accept trusted connections, as this setting cannot be changed later. For now, you can leave the list of accepted projects empty, and come back to add the NGINXaaS deployment project once created.

Author: I am going to lean towards leaving this out for now. We are recommending the secure option and the user has an option to choose the other one if they'd like to from the google docs which point them towards both.

>    - It is recommended to configure your network attachment to manually accept trusted connections, as this setting cannot be changed later. For now, you can leave the list of accepted projects empty, and come back to add the NGINXaaS deployment project once created.

Author: @amudukutore I pushed a small statement around development use-cases. Can you please take a look at it and let me know if this addresses your concerns?

>    - It is recommended to configure your network attachment to manually accept trusted connections, as this setting cannot be changed later. For now, you can leave the list of accepted projects empty, and come back to add the NGINXaaS deployment project once created.

Author: Resolving as pushed up all changes that were suggested in other comments.

### Create a network attachment

NGINXaaS requires a [network attachment](https://cloud.google.com/vpc/docs/about-network-attachments) to connect your NGINXaaS deployment to your VPC network.

1. Access the [Google Cloud Console](https://console.cloud.google.com/).
1. Create a consumer VPC network and subnetwork. See [Google's documentation on creating a VPC and subnet](https://cloud.google.com/vpc/docs/create-modify-vpc-networks#console_1) for a step-by-step guide.

>    - The region you choose in this step **must be the same** as the region where you plan to create your NGINXaaS deployment.

Reviewer 2: I think what you have here is fine but we may want to spell it out a bit further - perhaps something like:
_The region you select for the network attachment determines the region where your NGINXaaS deployment will be created. You do not manually select a region when creating an NGINXaaS deployment; it will automatically be created in the same region as the network attachment._

1. Create a network attachment in your new subnet. See [Google's documentation on creating a network attachment](https://cloud.google.com/vpc/docs/create-manage-network-attachments#create-network-attachments) for a step-by-step guide.

>    - We recommend you configure your network attachment to manually accept trusted connections for **production use cases**, as this setting cannot be changed later. For now, you can leave the list of accepted projects empty, and come back to add the NGINXaaS deployment project once created. You can choose to automatically accept connections on the network attachment for **development use cases**. If you choose to do so, you do not need to explicitly allow the NGINXaaS deployment project on the network attachment.

Author: 👍
